crates/typst-library/src/layout/flow.rs

```diff
@@ -436,6 +440,24 @@ impl<'a> FlowLayouter<'a> {
 
     /// Finish the frame for one region.
     fn finish_region(&mut self, vt: &mut Vt) -> SourceResult<()> {
+        if let Some(items) = self.last_frame_only_out_of_flow.take() {
+            self.items.splice(0..0, items);
+            self.finished.pop();
```

Comment:
I feel like I should push an empty frame here to replace the one I just removed, but doing so makes the last test I added fail (it adds an empty page between the text and the block). How does it know what region frames correspond to if I just remove frames randomly?